File: medplum-java/medplum-model/src/main/java/com/fhir/PlanDefinition_Condition.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutablePlanDefinition_Condition.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface PlanDefinition_Condition {
  Optional<List<Extension>> modifierExtension();

  Optional<List<Extension>> extension();

  Optional<String> id();

  Optional<Expression> expression();

  Optional<Plandefinition_conditionKind> kind();

  static ImmutablePlanDefinition_Condition.Builder builder() {
    return ImmutablePlanDefinition_Condition.builder();
  }
}

File: medplum-typescript/src/model/fhir/PlanDefinition_Condition.ts
import { Expression } from "./Expression";
import { Extension } from "./Extension";
import { Plandefinition_conditionKind } from "./Plandefinition_conditionKind";

export interface PlanDefinition_Condition {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    kind: Plandefinition_conditionKind | null | undefined;
    expression: Expression | null | undefined;
}
